update/genesis/export.go

```diff
@@ -317,6 +350,45 @@ func (se *stateExport) exportTx(key string, tx data.TransactionHandler) error {
 	return nil
 }
 
+func (se *stateExport) exportNodesSetupJson(validators map[uint32][]*state.ValidatorInfo) error {
+	acceptedListsForExport := []core.PeerType{core.EligibleList, core.WaitingList, core.JailedList}
+	initialNodes := make([]*sharding.InitialNode, 0)
+
+	for _, validatorsInShard := range validators {
+		for _, validator := range validatorsInShard {
+			if shouldExportValidator(validator, acceptedListsForExport) {
+				initialNodes = append(initialNodes, &sharding.InitialNode{
+					PubKey:        se.validatorPubKeyConverter.Encode(validator.GetPublicKey()),
+					Address:       se.addressPubKeyConverter.Encode(validator.GetRewardAddress()),
+					InitialRating: validator.GetRating(),
+				})
+			}
+		}
+	}
+
+	genesisNodesSetupHandler := se.genesisNodesSetupHandler
+	nodesSetup := &sharding.NodesSetup{
+		StartTime:                   genesisNodesSetupHandler.GetStartTime(),
+		RoundDuration:               genesisNodesSetupHandler.GetRoundDuration(),
+		ConsensusGroupSize:          genesisNodesSetupHandler.GetShardConsensusGroupSize(),
+		MinNodesPerShard:            genesisNodesSetupHandler.MinNumberOfShardNodes(),
+		ChainID:                     genesisNodesSetupHandler.GetChainId(),
+		MinTransactionVersion:       genesisNodesSetupHandler.GetMinTransactionVersion(),
+		MetaChainConsensusGroupSize: genesisNodesSetupHandler.GetMetaConsensusGroupSize(),
+		MetaChainMinNodes:           genesisNodesSetupHandler.MinNumberOfMetaNodes(),
+		Hysteresis:                  genesisNodesSetupHandler.GetHysteresis(),
+		Adaptivity:                  genesisNodesSetupHandler.GetAdaptivity(),
+		InitialNodes:                initialNodes,
+	}
+
+	nodesSetupBytes, err := json.MarshalIndent(nodesSetup, "", "  ")
+	if err != nil {
+		return err
+	}
+
+	return ioutil.WriteFile(filepath.Join(se.exportFolder, "nodesSetup.json"), nodesSetupBytes, 0755)
```

Comment:
Can you put the nodesSetup.json in a variable and use that everywhere?

Reply:
done

Reply:
ok